proto/substrait/extended_expression.proto

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+syntax = "proto3";
+
+package substrait;
+
+import "substrait/algebra.proto";
+import "substrait/extensions/extensions.proto";
+
+option csharp_namespace = "Substrait.Protobuf";
+option go_package = "github.com/substrait-io/substrait-go/proto";
+option java_multiple_files = true;
+option java_package = "io.substrait.proto";
+
+message ExpressionReference {
+  oneof expr_type {
+    Expression expression = 1;
+    AggregateFunction measure = 2;
+  }
+  NamedStruct base_schema = 2;
```

Comment:
I feel like maybe this should be stated at the extended expression level as opposed to per ExpressionReference. Thoughts?

Reply:
Correct. Multi expressions should evaluated on same input data thus one `base_schema ` will be okay. Will update it.